_posts/2020-04-03-css-grid.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: post
+title: Just Enough CSS Grid to be Dangerous
+date: 2020-04-03
+tags: css grid
+author: Tim Veletta
+image: img/blog/css-grid/named-areas.png
+---
+
+# Why CSS Grid?
+
+CSS Grid is used to specify the layout, size, position of HTML elements on a webpage by dividing it into major regions. Unlike `flexbox` which is a one-dimensional layout system, either rows _or_ columns, CSS Grid allows you to specify your layout in both rows _and_ columns.
+
+To properly understand the _need_ for something like CSS Grid I think its important to examine where we have come from in terms of web layouts. First we started using `<table>` for layout which became problematic as soon as we had build responsive web pages to support viewing on both mobile and desktop. Next came Bootstrap and, more importantly, the 12-column grid which is used in a majority of component libraries to this day.
+
+A key feature of Bootstrap was being able to specify elements which would occupy a variable number of columns based on the screen size; something that just wasn't easily possible with `<table>`. However this has also led to some rather messy HTML and "`<div>` hell".
+
+![<div> hell!](/img/blog/css-grid/div-hell.jpg)
+Image Credit: https://devrant.com/rants/877733/callback-hell-think-about-div-hell
+
+One of the real benefits of using CSS Grid is that it allows you produce much cleaner and more succinct HTML. Also by eliminating the 12-column grid we no longer force our content to conform to those columns making things such as a 5-column grid possible. One of the quotes that encapsulates CSS Grid in a great way is:
+
+> CSS Grid allows you to make HTML into what is was supposed to be. Markup of content. Not visuals, which belong in the CSS.
+
+Reference: https://hackernoon.com/how-css-grid-beats-bootstrap-85d5881cf163
```

Comment:
An inline example showing a complete mess of bootstrap relative div code, then removing the div-soup in favour of something extremely clean and semantic would be a good example, if you can find/write one.

A 'look at your code, now look at my code, look back your code, sadly, it is not my code' kind of thing.

Reply:
So CSS Grid is Old Spice Guy, got it!